src/cryptoadvance/specter/server_endpoints/controller.py

```diff
@@ -170,45 +166,35 @@ def node_setup_wizard(step):
 @app.route("/setup_bitcoind", methods=["POST"])
 @login_required
 def setup_bitcoind():
-    bitcoind_setup_status = app.specter.config.get(
-        "bitcoind_setup", {"stage_progress": -1}
-    )
     app.specter.config["rpc"]["datadir"] = request.form.get(
         "bitcoin_core_datadir", app.specter.config["rpc"]["datadir"]
     )
     app.specter._save()
     if (
         not os.path.isfile(app.specter.bitcoind_path)
-        and bitcoind_setup_status["stage_progress"] == -1
+        and app.specter.setup_status["bitcoind"]["stage_progress"] == -1
     ):
-        app.specter.config["bitcoind_setup"]["stage_progress"] = 0
-        app.specter.config["bitcoind_setup"]["error"] = ""
-        app.specter._save()
+        app.specter.update_setup_status("bitcoind", "Strarting Bitcoin Core Setup", 0)
         t = threading.Thread(
             target=setup_bitcoind_thread,
             args=(app.specter, app.config["INTERNAL_BITCOIND_VERSION"]),
         )
         t.start()
     elif os.path.isfile(app.specter.bitcoind_path):
         return {"error": "bitcoind already installed"}
-    elif bitcoind_setup_status["stage_progress"] != -1:
+    elif app.specter.setup_status["bitcoind"]["stage_progress"] != -1:
         return {"error": "bitcoind installation is still under progress"}
     return {"success": "Starting Bitcoin Core setup!"}
 
 
 @app.route("/setup_bitcoind_directory", methods=["POST"])
 @login_required
 def setup_bitcoind_directory():
-    bitcoind_setup_status = app.specter.config.get(
-        "bitcoind_setup", {"stage_progress": -1}
-    )
     if (
         os.path.isfile(app.specter.bitcoind_path)
-        and bitcoind_setup_status["stage_progress"] == -1
+        and app.specter.setup_status["bitcoind"]["stage_progress"] == -1
     ):
-        app.specter.config["bitcoind_setup"]["stage_progress"] = 0
-        app.specter.config["bitcoind_setup"]["error"] = ""
-        app.specter._save()
+        app.specter.update_setup_status("bitcoind", "Strarting Bitcoin Core Setup", 0)
```

Comment:
There is a typo in there. Ideally, the states (stati?)  are CONSTANTS imported and centrally defined. So can you replace with contsants? That would make such typos-issue also impossible.